packages/AdaptiveCards/Actions/SendActionExecuteResponse.cs

```diff
@@ -0,0 +1,125 @@
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Components.AdaptiveCards
+{
+    public class SendActionExecuteResponse : Dialog
+    {
+        [JsonConstructor]
+        public SendActionExecuteResponse([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            // enable instances of this command as debug break point
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+        }
+
+        [JsonProperty("$kind")]
+        public const string Kind = "Microsoft.Bot.Components.SendActionExecuteResponse";
+
+        [JsonProperty("disabled")]
+        public BoolExpression Disabled { get; set; }
+
+        [JsonProperty("statusCode")]
+        public IntExpression StatusCode { get; set; } = new IntExpression(200);
+
+       [JsonConverter(typeof(StringEnumConverter), /*camelCase*/ true)]
+        public enum ValueType
+        {
+            /// <summary>
+            /// The response includes an Adaptive Card
+            /// </summary>
+            AdaptiveCard,
+
+            /// <summary>
+            /// The response includes a message
+            /// </summary>
+            Message
+        }
+
+        [JsonProperty("type")]
+        public EnumExpression<ValueType> Type { get; set; } = new EnumExpression<ValueType>(ValueType.Message);
+
+        [JsonProperty("value")]
+        public ValueExpression Value { get; set; }
+
+        public async override Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (options is CancellationToken)
+            {
+                throw new ArgumentException($"{nameof(options)} cannot be a cancellation token");
+            }
+
+            if (this.Disabled != null && this.Disabled.GetValue(dc.State) == true)
+            {
+                return await dc.EndDialogAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+
+            if (dc.Context.Activity.Type != ActivityTypes.Invoke || dc.Context.Activity.Name != "adaptiveCard/action")
+            {
+                throw new Exception($"{this.Id}: should only be called in repsonse to 'invoke' activities with a name of 'adaptiveCard/action'.");
+            }
+
+            // Get parameters
+            var statusCode = StatusCode?.GetValue(dc.State);
+            var type = Type?.GetValue(dc.State);
+            var value = Value?.GetValue(dc.State);
+            string mimeType;
+
+            // Validate params
+            if (statusCode != null && statusCode >= 200 && statusCode < 300)
+            {
+                switch (type)
+                {
+                    case ValueType.Message:
+                        mimeType = "application/vnd.microsoft.activity.message";
```

Comment:
Are these not defined as consts anywhere in the SDK layer? If not, probably worth defining them in some common location in this lib to prevent duplication of the raw values.

Reply:
They will eventually be defined but these are brand new types. Once they get defined I'll migrate. Adding a TODO

---
In reply to: [610215452](https://github.com/microsoft/botframework-components/pull/862#discussion_r610215452) [](ancestors = 610215452)